### PR TITLE
chore: Add the 4.1 release notes

### PR DESCRIPTION
**Pull request by sadpandajoe**
### SUMMARY

This PR adds the 4.1 release notes

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: RELEASING/release-notes-4-1/README.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+under the License.
+-->
+
+# Release Notes for Superset 4.1.0
+
+Superset 4.1.0 brings a range of new features and quality of life improvements.  This release is a minor version, meaning it doesn't include any breaking changes to ensure a seamless transition for our users. Here are some of the highlights of this release.
+
+### Big Number With Time Period Updates
+
+We released a [Big Number with Comparison](https://github.com/apache/superset/pull/26908) chart as part of Superset 4.0. With the latest update, there are now [color options](https://github.com/apache/superset/pull/27524) for comparisons. The chart now also uses [standardize controls](https://github.com/apache/superset/pull/27193) such that when switching charts will maintain the selected metrics. To enable the new chart, you'll need to enable the `CHART_PLUGINS_EXPERIMENTAL` feature flag.
```
Comment (yousoph) [inline RELEASING/release-notes-4-1/README.md:26]:
```suggestion
We released a [Big Number with Time Period Comparison](https://github.com/apache/superset/pull/26908) chart as part of Superset 4.0. With the latest update, there are now [color options](https://github.com/apache/superset/pull/27524) for comparisons. The chart now also uses [standardize controls](https://github.com/apache/superset/pull/27193) such that when switching charts will maintain the selected metrics. To enable the new chart, you'll need to enable the `CHART_PLUGINS_EXPERIMENTAL` feature flag.
```

File: RELEASING/release-notes-4-1/README.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+<div>
+    <image src="media/big_number_chart.png" alt="Image" width="100%">
+</div>
+
+### Time Comparisson With Table
```
Comment (yousoph) [inline RELEASING/release-notes-4-1/README.md:32]:
```suggestion
### Table with Time Comparison
```

**Comment by michael-s-molina** [comment]
Thanks for this @sadpandajoe. The Histogram screenshot is showing the legacy Histogram. Feel free to use the images available at https://github.com/apache/superset/pull/28652.

Did we already cut 4.1? I was not able to find the branch at https://github.com/apache/superset/tree/4.1

**Comment by michael-s-molina** [comment]
> @michael-s-molina we did not, but we are getting close to cutting it

Got it. Let me know when it's available so we can do a compare with 4.0 to check possible features that should be included in the release notes.

**Comment by michael-s-molina** [comment]
Wow, I was going to write the migration paragraph but you already did! Thank you so much @sadpandajoe!

**Comment by michael-s-molina** [comment]
Now that [4.1](https://github.com/apache/superset/tree/4.1) was cut, I compared it with 4.0 with the focus on features to see what else is missing in the release notes:

```
git log origin/4.0...origin/4.1 --pretty=format:%s --grep=feat
```

Features that we might want to highlight:

```
feat: add slackv2 notification (#29264)
feat: add connector for CouchbaseDB (#29225)
feat(database): Add OceanBase support (#29496)
feat: Adds the ECharts Sankey chart (#29329)
feat(table): Table with Time Comparison (#28057)
feat: Adds the ECharts Histogram chart (#28652)
feat: add support for catalogs (#28416)
feat(maps): Adding ALL the countries to the Country Map plugin! 🌎 (#28265)
feat(dashboard): Add metadata bar to the header (#27857)
feat(explore): Color scheme groups, new color schemes (#27995)
feat: Utility function to render chart tooltips (#27950)
feat: custom refresh frequency (#24449)
feat: new Excel upload form and API (#28105)
feat: new CSV upload form and API (#27840)
feat: Slack Avatar integration (#27849)
feat(viz picker): Remove some tags, refactor Recommended section (#27708)
feat: Adds the ECharts Heatmap chart (#25353)
feat: Adds option to disable drill to detail per database (#27536)
feat(embedded-sdk): Add 'urlParams' option to pass query parameters to embedded dashboard (#24408)
```

The remaining features:
```
feat(frontend/hooks): replace 3rd-party BroadcastChannel with native Web API equivalence (#29584)
feat: custom values to sandbox iframe (#29590)
feat(build): uplift Lerna + replace insecure shortid with nanoid + uplift Yeoman-related 
packages + ESM-ize generator-superset (#29419)
feat(build): uplift Storybook to v8 (#29408)
feat: add support to NOT LIKE operator (#29384)
feat: Enable customizing the docker admin password (#29498)
feat(dashboard): add API endpoints for generating and downloading screenshots (#29187)
feat(CLI command): Apache Superset "Factory Reset" CLI command #27207 (#27221)
feat: Add Ant Design 5 Theme  (#29328)
feat(e2e): implementing Cypress Dashboard on `master` branch merges (#29351)
feat: Adds chart IDs option to migrate-viz (#29361)
feat(build): uplift `Jest` to v29 (#29118)
feat: add new SQLLAB_FORCE_RUN_ASYNC feature flag (#29231)
feat(dashboard): Enables pivot table download option at dashboard level (#29123)
feat: Dashboard tabs api endpoint (#27962)
feat: Improves the Drill By feature (#29242)
feat: Support a dynamic minimum interval for alerts and reports (#29241)
feat(trino): Add functionality to upload data (#29164)
feat(echarts-pie): add string template support for labels (#28774)
feat(formatters): Add custom d3-time-format locale (#24263)
feat: OAuth2 client initial work (#29109)
feat: add Current time-range options for time filter (#28637)
feat: Adds Histogram chart migration logic (#28780)
feat(helm): allow removal of Node & Worker replicas for custom HPA solutions (#28762)
feat: Adds the Featured Charts dashboard (#28789)
feat: impersonate with email prefix (#28770)
feat: bake translations as part of the build processes (#28483)
feat(reports): allowing the email mutator to update recipients (#27851)
feat: add Nightingale chart support for echarts pie chart (#28597)
feat: Adds Bar chart migration logic (#28602)
feat: unpack payload into log function (#28521)
feat: Data Zoom scrolls using the mouse (mark II) (#28629)
feat: Customizable email subject name (#26327)
fix(docs): add missing link to meta-cross-db feature flag docs (#28433)
feat: clarify that 'Text' supports markdown (#28425)
feat: catalog support for Databricks native (#28394)
feat(SIP-95): catalogs in SQL Lab and datasets (#28376)
feat(reports): Set a minimum interval for each report's execution (#28176)
feat(docs): uplift Docusaurus to v3 (#28345)
feat(SIP-95): permissions for catalogs (#28317)
feat: new Columnar upload form and API (#28192)
feat: accelerate webpack builds with filesystem cache (#28282)
feat: Add Czech Republic country map. (#28035)
feat(country-map): Adds Philippines regional map and updates/cleans existing 
Philippines provincial map (#27933)
feat(translations): Traditional Chinese translation files added (#28169)
feat(SIP-95): new endpoint for table metadata (#28122)
feat: improve event logging for queries + refactor (#27943)
feat: label PR with release tags (#28107)
feat(sip-95): new endpoint for extra table metadata (#28063)
chore(docs): splitting out "stable" feature flags by intent (config vs feature dev) (#28016)
feat(dbview): Add token request button to DuckDB and MotherDuck database modal (#27908)
feat: optimize docker-compose up for faster boot time (#27953)
feat: add option to disable rendering of html in sql lab and table chart (#27969)
feat(alert report tabs): adding feature flag (#27773)
feat: GHA to bump python packages using supersetbot (#27863)
feat(explore): Clear temporal filter value (#27788)
feat(accessibility): add tabbing to chart menu in dashboard (#26138)
feat: move supersetbot out of repo (#27647)
feat(SIP-85): OAuth2 for databases (#27631)
feat: setup a pyproject.toml (#27859)
feat(db): Adding DB_SQLA_URI_VALIDATOR (#27847)
feat: Adds Heatmap chart migration logic (#27771)
feat(db_engine): Add custom_user_agent when connecting to MotherDuck (#27665)
feat: use the local supersetbot (#27615)
feat(jinja): metric macro (#27582)
feat(alerts-reports): adding pdf filetype to email and slack reports (#27497)
feat: support for KQL in `SQLScript` (#27522)
feat(bar_chart): Stacked Bar chart with Time comparison in separated stacks (#27589)
feat(supersetbot): label PRs and issues with author's public org (#27571)
feat(maps): Add Italy regions code to the map generator notebook (#27542)
feat: `improve _extract_tables_from_sql` (#26748)
feat: `improve _extract_tables_from_sql` (#26748)
feat(plugins):  add color options for big number with time comparison (#27524)
feat: Add Turkey's regions to country map visualization (#27455)
feat(supersetbot): introduce `supersetbot` as its own npm package, CLI and comment-operated bot (#27046)
feat: show more information when loading chart (#27255)
feat: docker-compose to work off repo Dockerfile (#27434)
feat(translations): Turkish translation files added (#27244)
feat: Add repo activity stats to README.md (#27372)
feat: Responsive UI for Big Number with Time Comparison (#27375)
feat: support to fetch multiple date time in time_range endpoint (#27370)
feat: datediff in datetime_parser (#27368)
feat(logs context): Adding dashboard id to logs context (#27298)
feat(jinja): current_user_email macro (#27197)
feat(ci): no more docker builds on PR-related events (#27146)
feat: Use standardized controls in Big Number with Time Comparison (#27193)
feat(docs): Adds an "Edit this page on GitHub" button to docs pages (#27176)
feat(helm): optionally set pod disruption budgets (#27163)
feat(adt): add 403 to api response status codes (#27162)
```

Feel free to use the same command altering `grep` if you want to check for fixes or chores that might be worth highlighting.

**Comment by sadpandajoe** [comment]
> Now that [4.1](https://github.com/apache/superset/tree/4.1) was cut (thank you), I compared it with 4.0 with the focus on features to see what else is missing in the release notes:
> 
> ```
> git log origin/4.0...origin/4.1 --pretty=format:%s --grep=feat
> ```
> 
> Features that we might want to highlight:
> 
> ```
> feat: add slackv2 notification (#29264)
> feat: add connector for CouchbaseDB (#29225)
> feat(database): Add OceanBase support (#29496)
> feat: Adds the ECharts Sankey chart (#29329)
> feat(table): Table with Time Comparison (#28057)
> feat: Adds the ECharts Histogram chart (#28652)
> feat: add support for catalogs (#28416)
> feat(maps): Adding ALL the countries to the Country Map plugin! 🌎 (#28265)
> feat(dashboard): Add metadata bar to the header (#27857)
> feat(explore): Color scheme groups, new color schemes (#27995)
> feat: Utility function to render chart tooltips (#27950)
> feat: custom refresh frequency (#24449)
> feat: new Excel upload form and API (#28105)
> feat: new CSV upload form and API (#27840)
> feat: Slack Avatar integration (#27849)
> feat(viz picker): Remove some tags, refactor Recommended section (#27708)
> feat: Adds the ECharts Heatmap chart (#25353)
> feat: Adds option to disable drill to detail per database (#27536)
> feat(embedded-sdk): Add 'urlParams' option to pass query parameters to embedded dashboard (#24408)
> ```
> 
> The remaining features:
> 
> ```
> feat(frontend/hooks): replace 3rd-party BroadcastChannel with native Web API equivalence (#29584)
> feat: custom values to sandbox iframe (#29590)
> feat(build): uplift Lerna + replace insecure shortid with nanoid + uplift Yeoman-related 
> packages + ESM-ize generator-superset (#29419)
> feat(build): uplift Storybook to v8 (#29408)
> feat: add support to NOT LIKE operator (#29384)
> feat: Enable customizing the docker admin password (#29498)
> feat(dashboard): add API endpoints for generating and downloading screenshots (#29187)
> feat(CLI command): Apache Superset "Factory Reset" CLI command #27207 (#27221)
> feat: Add Ant Design 5 Theme  (#29328)
> feat(e2e): implementing Cypress Dashboard on `master` branch merges (#29351)
> feat: Adds chart IDs option to migrate-viz (#29361)
> feat(build): uplift `Jest` to v29 (#29118)
> feat: add new SQLLAB_FORCE_RUN_ASYNC feature flag (#29231)
> feat(dashboard): Enables pivot table download option at dashboard level (#29123)
> feat: Dashboard tabs api endpoint (#27962)
> feat: Improves the Drill By feature (#29242)
> feat: Support a dynamic minimum interval for alerts and reports (#29241)
> feat(trino): Add functionality to upload data (#29164)
> feat(echarts-pie): add string template support for labels (#28774)
> feat(formatters): Add custom d3-time-format locale (#24263)
> feat: OAuth2 client initial work (#29109)
> feat: add Current time-range options for time filter (#28637)
> feat: Adds Histogram chart migration logic (#28780)
> feat(helm): allow removal of Node & Worker replicas for custom HPA solutions (#28762)
> feat: Adds the Featured Charts dashboard (#28789)
> feat: impersonate with email prefix (#28770)
> feat: bake translations as part of the build processes (#28483)
> feat(reports): allowing the email mutator to update recipients (#27851)
> feat: add Nightingale chart support for echarts pie chart (#28597)
> feat: Adds Bar chart migration logic (#28602)
> feat: unpack payload into log function (#28521)
> feat: Data Zoom scrolls using the mouse (mark II) (#28629)
> feat: Customizable email subject name (#26327)
> fix(docs): add missing link to meta-cross-db feature flag docs (#28433)
> feat: clarify that 'Text' supports markdown (#28425)
> feat: catalog support for Databricks native (#28394)
> feat(SIP-95): catalogs in SQL Lab and datasets (#28376)
> feat(reports): Set a minimum interval for each report's execution (#28176)
> feat(docs): uplift Docusaurus to v3 (#28345)
> feat(SIP-95): permissions for catalogs (#28317)
> feat: new Columnar upload form and API (#28192)
> feat: accelerate webpack builds with filesystem cache (#28282)
> feat: Add Czech Republic country map. (#28035)
> feat(country-map): Adds Philippines regional map and updates/cleans existing 
> Philippines provincial map (#27933)
> feat(translations): Traditional Chinese translation files added (#28169)
> feat(SIP-95): new endpoint for table metadata (#28122)
> feat: improve event logging for queries + refactor (#27943)
> feat: label PR with release tags (#28107)
> feat(sip-95): new endpoint for extra table metadata (#28063)
> chore(docs): splitting out "stable" feature flags by intent (config vs feature dev) (#28016)
> feat(dbview): Add token request button to DuckDB and MotherDuck database modal (#27908)
> feat: optimize docker-compose up for faster boot time (#27953)
> feat: add option to disable rendering of html in sql lab and table chart (#27969)
> feat(alert report tabs): adding feature flag (#27773)
> feat: GHA to bump python packages using supersetbot (#27863)
> feat(explore): Clear temporal filter value (#27788)
> feat(accessibility): add tabbing to chart menu in dashboard (#26138)
> feat: move supersetbot out of repo (#27647)
> feat(SIP-85): OAuth2 for databases (#27631)
> feat: setup a pyproject.toml (#27859)
> feat(db): Adding DB_SQLA_URI_VALIDATOR (#27847)
> feat: Adds Heatmap chart migration logic (#27771)
> feat(db_engine): Add custom_user_agent when connecting to MotherDuck (#27665)
> feat: use the local supersetbot (#27615)
> feat(jinja): metric macro (#27582)
> feat(alerts-reports): adding pdf filetype to email and slack reports (#27497)
> feat: support for KQL in `SQLScript` (#27522)
> feat(bar_chart): Stacked Bar chart with Time comparison in separated stacks (#27589)
> feat(supersetbot): label PRs and issues with author's public org (#27571)
> feat(maps): Add Italy regions code to the map generator notebook (#27542)
> feat: `improve _extract_tables_from_sql` (#26748)
> feat: `improve _extract_tables_from_sql` (#26748)
> feat(plugins):  add color options for big number with time comparison (#27524)
> feat: Add Turkey's regions to country map visualization (#27455)
> feat(supersetbot): introduce `supersetbot` as its own npm package, CLI and comment-operated bot (#27046)
> feat: show more information when loading chart (#27255)
> feat: docker-compose to work off repo Dockerfile (#27434)
> feat(translations): Turkish translation files added (#27244)
> feat: Add repo activity stats to README.md (#27372)
> feat: Responsive UI for Big Number with Time Comparison (#27375)
> feat: support to fetch multiple date time in time_range endpoint (#27370)
> feat: datediff in datetime_parser (#27368)
> feat(logs context): Adding dashboard id to logs context (#27298)
> feat(jinja): current_user_email macro (#27197)
> feat(ci): no more docker builds on PR-related events (#27146)
> feat: Use standardized controls in Big Number with Time Comparison (#27193)
> feat(docs): Adds an "Edit this page on GitHub" button to docs pages (#27176)
> feat(helm): optionally set pod disruption budgets (#27163)
> feat(adt): add 403 to api response status codes (#27162)
> ```
> 
> Feel free to use the same command altering `grep` if you want to check for fixes or chores that might be worth highlighting.

@yousoph what do you think? Which additional ones do we want to highlight. Ones we have so far
* Big Number with Time Period
* table with time comparison
* heatmap v2
* histogram v2
* upload forms for csv, excel and columnar
* oauth2 for databases
* catalog support for databases

**Comment by yousoph** [comment]
I'd add Sankey to the list you have right now @sadpandajoe

**Comment by sadpandajoe** [comment]
> I'd add Sankey to the list you have right now @sadpandajoe

@yousoph added and also a blurb about slack v2 that probably needs to be reviewed.

**Review by sfirke — APPROVED** [review]
This is a great start.  I agree with Michael that https://github.com/apache/superset/pull/27950, that # and % both appear in tooltips, is worth highlighting.  The metadata bar on top of dashboard (https://github.com/apache/superset/pull/27857) looks like it will display in embedded and standalone modes, which might be worth calling out so users of those features are aware.

https://github.com/apache/superset/pull/27497 and https://github.com/apache/superset/pull/28637 are not huge features code-wise but were often requested by the community. Those might be worth highlighting too but feel free to ignore if that makes the list too long.

**Comment by sadpandajoe** [comment]
> This is a great start. I agree with Michael that #27950, that # and % both appear in tooltips, is worth highlighting. The metadata bar on top of dashboard (#27857) looks like it will display in embedded and standalone modes, which might be worth calling out so users of those features are aware.
> 
> #27497 and #28637 are not huge features code-wise but were often requested by the community. Those might be worth highlighting too but feel free to ignore if that makes the list too long.

@sfirke I've added the changes requested in the latest commit.

**Comment by sadpandajoe** [comment]
🏷️ 4.1

**Comment by sadpandajoe** [comment]
@supersetbot label 4.1

**Comment by mistercrunch** [comment]
@sadpandajoe wondering if you saw how dosu can help with release notes (?)

**Comment by sadpandajoe** [comment]
> @sadpandajoe wondering if you saw how dosu can help with release notes (?)

No, I didn't know it can do that. I looked on their website (dosu.dev?) and didn't see anything on documentation of commands.

**Review by yousoph — APPROVED** [review]

**Pull request merged**